### PR TITLE
[FLINK-28861][table][docs] Update "upgrading" documentation to includ…

### PR DESCRIPTION
**Pull request by dannycranmer**
…e notice regarding the Table API operator issue

## What is the purpose of the change

Update documentation to add a notice regarding the Table API operator issue in FLINK-28861

## Brief change log

- Updated "upgrading" documentation

## Verifying this change

- Visualised locally in IDE

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: docs/content.zh/docs/ops/upgrading.md [hunk 1]
```diff
@@ -518,7 +518,12 @@ Savepoints are compatible across Flink versions as indicated by the table below:
           <td class="text-center"></td>
           <td class="text-center"></td>
           <td class="text-center">O</td>
-          <td class="text-left"></td>
+          <td class="text-left">
+            Due to FLIP-190, Flink 1.15.0/1.15.1 changed the way UIDs are generated for TableAPI
```
Comment (twalthr) [inline docs/content.zh/docs/ops/upgrading.md:522]: Maybe we can use the release notes that I added in the JIRA issue (slightly adapted):
```
For Table API: 1.15.0 and 1.15.1 generated non-deterministic UIDs for operators that make it difficult/impossible to restore state or upgrade to next patch version. A new table.exec.uid.generation config option (with correct default behavior) disables setting a UID for new pipelines from non-compiled plans. Existing pipelines can set table.exec.uid.generation=ALWAYS if the 1.15.0/1 behavior was acceptable due to a stable environment.
```

I tried to be as concise and helpful as possible.

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks for updating this section @dannycranmer. I tried to make it a bit more precise. The old behavior was not causing issues for every user. As long as the static counter starts at 0 (stable environment), the state can still be restored. This is why the ALWAYS strategy still exists.

File: docs/content.zh/docs/ops/upgrading.md (hunk 1 above)
Comment (twalthr) [reply docs/content.zh/docs/ops/upgrading.md:522]: Feel free to shorten it if needed for the table.

Comment (dannycranmer) [reply docs/content.zh/docs/ops/upgrading.md]: Thanks for the suggestion. I have updated it according to your recommendation, with the addition of a link to the Jira. I do not think it needs shortening, it is not tallest cell in the table. We can always shorten it in the future when it is less relevant.

**Review by twalthr — APPROVED** [review]
Thanks for the update @dannycranmer. LGTM.

**Pull request merged**